src/http.ts

```diff
@@ -24,6 +26,14 @@ export interface HttpMessage {
 export interface HttpRequest extends HttpMessage {
   method: HttpMethod;
   url: string;
```

Comment:
It looks like the [nodejs `URL` class](https://nodejs.org/dist/latest-v10.x/docs/api/url.html) is compatible with [whatwg URL class](https://developer.mozilla.org/en-US/docs/Web/API/URL). I have tried it in node console and apparently it is working:
```
> const url = new URL("https://user:pass@sub.example.com:8080/p/a/t/h?query=string#hash")
undefined
> url
URL {
  href:
   'https://user:pass@sub.example.com:8080/p/a/t/h?query=string#hash',
  origin: 'https://sub.example.com:8080',
  protocol: 'https:',
  username: 'user',
  password: 'pass',
  host: 'sub.example.com:8080',
  hostname: 'sub.example.com',
  port: '8080',
  pathname: '/p/a/t/h',
  search: '?query=string',
  searchParams: URLSearchParams { 'query' => 'string' },
  hash: '#hash' }
```

The class has the following features:
* parsing the url and provide fields for every part of the url.
* provide the querystring handling methods using [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams).

Please make sure we cannot use this class and if we could, please consider using it.

Reply:
I have tried this idea and looks like we can somehow figure out the whole url:
``` js
    const originalUrl = `http://${nodeReq.headers["host"]}${nodeReq.url}`;
```